src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx

```diff
@@ -1007,10 +1010,144 @@ XclBinUtilities::createMemoryBankGrouping(XclBin & xclbin)
+
+  // currently transform_static prints lots of messages to console (stdout)
+  // redirect the output to a file, so that console looks cleaner
+  int sout = dup(fileno(stdout));
+  FILE* f = freopen("/dev/null", "w", stdout);
+  if (f == nullptr)
+    std::cout << "stdout redirect failed" << std::endl;
+
+  XUtil::TRACE("Transform " + fileName);
+  int returnCode = std::system(command.c_str());
+
+  // restore stdout
+  dup2(sout,fileno(stdout));
+  close(sout);
+
+  // std::cout << "tranform_static returns " << returnCode << std::endl;
+  // returnCode is 0 for success, non-0 for error
+  return returnCode;
+}
+
+void 
+XclBinUtilities::transformAiePartitionPDIs(XclBin & xclbin)
+{
+  // find all sections with type "AIE_PARTITION" in xclbin
+  // create a temp empty folder on disk, e.g. "ap_temp"
+  // for each section
+  //   dump the content (pdi files) to the temp folder
+  //      <index name>/
+  //          orig/
+  //             aie_partition.json
+  //             1.pdi, 2.pdi ...
+  //   copy orig/ to transform/          
+  //   for each pdi in transform/, call trasform_static
+  //      <index name>/
+  //          transform/
+  //             aie_partition.json
+  //             transformed 1.pdi, 2.pdi ...
+  //   construct the string for sections to be removed, add to container A
+  //   construct PSD for the sections in <temp>/<index name>/transform, add to container B
+  // }
+  // for each section in container A
+  //   removeSection(PSD)
+  // for each PSD in container B
+  //   addSection(PSD)
+  // delete the temp dir
+
+  // create a temp directory
+  // fs::current_path or fs::temp_directory_path?
+  std::string apJson = "aie_partition.json";
+  // ap: aie_partition
+  fs::path tempDir = fs::current_path() / "ap_temp";
+
+  std::vector<std::string> removeSections;
+  std::vector<std::string> addSections;
+  std::vector<Section*> sections = xclbin.findSection(AIE_PARTITION, true);
+  for (auto& pSection : sections) {
+    // get the section name, index
+    std::string sSectionKind = pSection->getSectionKindAsString();
+    std::string sSectionIndex = pSection->getSectionIndexName();
+    std::string sSectionName = pSection->getName();
+    // std::cout << "sSectionKind = " << sSectionKind << std::endl;
+    // std::cout << "sSectionIndex = " << sSectionIndex << std::endl;
+    // std::cout << "sSectionName = " << sSectionName << std::endl;
+
+    // note sSectionIndex could be an empty string
+    fs::path origDir = tempDir / sSectionIndex / "orig";
+    fs::path transformDir = tempDir / sSectionIndex / "transform";
+    fs::path origApJsonPath = origDir / apJson;
+    fs::path tranApJsonPath = transformDir / apJson;
+    // std::cout << "origDir = " << origDir.string() << std::endl;
+
+    try {
+      fs::create_directories(origDir);
+    }
+    catch (std::exception& e) { // Not using fs::filesystem_error since std::bad_alloc can throw too.
+        std::cout << e.what() << std::endl;
```

Comment:
And what? Life goes on?

Reply:
add throw

Reply:
But at the end, this is against the lazy philosophy of using exception.
If you cannot add any value by adding a try/catch, just don't. Someone else will catch it upstairs or if nobody cares, we do not care. :-)